### ISSUE TITLE
Add notifications

### ISSUE THREAD

**Issue by enjeck** [nextcloud/tables#1460]
It would be nice to have a way to activate a notification if content is introduced. 

Even better, if new content is available on a view (which takes advantages of the filters), to generate a notification to specific contacts/groups

**Comment by ISC-PRO** [comment]
Releated with #552.

**Comment by blizzz** [comment]
Notifications would integration with the Notifications and Activity system in Nextcloud (cf screenshot). We should offer to issue those for these cases:

- A row was created/updated/deleted in a table or view I am managing
- A row was created/updated/deleted in any accessible table or view
- I was added in an accessible table row¹
- One of my groups was added in an accessible table row¹

Perhaps also

- A table or view was shared with me
- A table or view was modified²
- My permissions were modified³

¹ refers to the UserGroupColumn (and possibly Contacts field 🤔 )
² structurally, not content-wise
³ e.g. promotion or demotion as manager

![Image](https://github.com/user-attachments/assets/95d4cc92-422e-44df-a6ea-d91b39a69b1d)

**Comment by luka-nextcloud** [comment]
> For push notifications, it shall be possible to enable them per view/tables. Need to discuss precise granularity yet.

It is reasonable to me. As I understand, to achieve this, we might need to create a new table for storing notification config per view/tables.

**Comment by luka-nextcloud** [comment]
@blizzz I have checked again. I think we can use user config (`IUserConfig`) for storing notify config per table/view instead. Regarding notification events, we should define 2 groups:

- General (in `/settings/user/notifications`):
  - Table or view changed (create, edit, delete)
  - Sharing (table or view was shared with me)

- Per table/view:
  - Row changed (create, edit, delete)
  - Column changed (create, edit, delete)
  - Mentioned (added in UserGroup cell)

BTW, notification config per table/view can be placed in the sidebar details tab.

<img width="729" height="357" alt="Image" src="https://github.com/user-attachments/assets/30df749f-acab-4061-bb05-d1147786e285" />

**Comment by blizzz** [comment]
- not sure whether iuserconfig is the best place, but it should work, and could be a start
- it should be considered to remove the setting, when a table or view renders unavailable… or it can stay there, because
- it should be checked before sending out a notification, whether the table or view is still available to the user (at least runtime caching is welcome)
- perhaps easier (especially UI-wise)  to combine row changed and mentioned, and only have a bell to tick on or off. This could be then in the sidebar as well
  - In a second iteration, we can bring fine-grained controls to the sidebar if it turns out they are actually needed/wanted by users
- stretch goal/follow up: mention within in richtext (if mentioning actually that works there atm 🤔 - if not, then this is irrelevant)


@marcoambrosini what do you think?

**Comment by luka-nextcloud** [comment]
> - it should be considered to remove the setting, when a table or view renders unavailable… or it can stay there, because
> - it should be checked before sending out a notification, whether the table or view is still available to the user (at least runtime caching is welcome)

Agree, activity/notification events should only be sent to the affected users who still have access to table/view

**Comment by marcoambrosini** [comment]
I think that the best place for this are the current "Edit table" and "Edit view" entries, which I would rename into "Table settings" and "View settings".

<img width="171" height="389" alt="Image" src="https://github.com/user-attachments/assets/cc71c903-279e-4f29-a520-834e6479bc00" />

It would also be nice if this menu was duplicated to the top right of the content


> combine row changed and mentioned

If it's not too much I would separate "All changes" from "Mentions"

**Comment by luka-nextcloud** [comment]
@marcoambrosini 
> I think that the best place for this are the current "Edit table" and "Edit view" entries, which I would rename into "Table settings" and "View settings".

I don't think so. The notifications setting is per user per table/view but the "Edit table" and "Edit view" are only for the table owner.

**Comment by marcoambrosini** [comment]
Can't we introduce this dialog for users too and just remove the non owner sections from the regular user's view? Something similar the conversation settings in Talk. What do you think?

**Comment by estux** [comment]
Hello, don't know if this is the right issue for this but I think that notifications should also be triggered e.g. when e.g. you have date fields to check in advance (days/weeks before) to update data in the same row (I'm thinking about managing membership list and renewal for an association...where you have the date of the annual payment as a field to check to ask for the renewal in advance).

Thanks for your work and for considering it!
